sarracen/interpolate.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+
+
+def interpolate2D(data, x, y, target, kernel, xmin, ymin, pixwidthx, pixwidthy, pixcountx, pixcounty):
+    """
+    Interpolates particle data in a SarracenDataFrame across two directional axes to a 2D
+    grid of pixels.
+
+    :param data: The particle data, in a SarracenDataFrame.
+    :param x: The column label of the x-directional axis.
+    :param y: The column label of the y-directional axis.
+    :param target: The column label of the target smoothing data.
+    :param kernel: The kernel to use for smoothing the target data.
+    :param xmin: The starting x-coordinate (in particle data space).
+    :param ymin: The starting y-coordinate (in particle data space).
+    :param pixwidthx: The width that each pixel represents in particle data space.
+    :param pixwidthy: The height that each pixel represents in particle data space
+    :param pixcountx: The number of pixels in the output image in the x-direction.
+    :param pixcounty: The number of pixels in the output image in the y-direction.
+    :return: The output image, in a 2-dimensional numpy array.
+    """
```

Comment:
We can add some error checks on the input values here. Two simple ones are checks on the pixel width and count, as both should be positive. Can raise a ValueError() if they are 0 or negative.